server/src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -2802,7 +2803,9 @@ protected void commitIndexWriter(final IndexWriter writer, final Translog transl
                  * {@link IndexWriter#commit()} call flushes all documents, we defer computation of the maximum sequence number to the time
                  * of invocation of the commit data iterator (which occurs after all documents have been flushed to Lucene).
                  */
-                final Map<String, String> commitData = Maps.newMapWithExpectedSize(8);
+                final Map<String, String> extraCommitUserData = getCommitExtraUserData();
+                final Map<String, String> commitData = Maps.newMapWithExpectedSize(8 + extraCommitUserData.size());
+                commitData.putAll(extraCommitUserData);
```

Comment:
Is there any upper bound for commit data? Any chances that we blow up that buffer if we allow potentially unlimited data in?

Reply:
I don't see an upper bound, but I'm also not sure I am concerned, as this is not user-provided, in the sense that it's code/plugins (extension of the InternalEngine) that will put any custom data. We should probably make sure ourselves we do not blow it up by putting unreasonable custom data. The limit seems to be memory or disk size. But I would not mind putting some sort of limit.

Reply:
This is set by extenders of the `InternalEngine`. Adding user-data tends to be a deliberate decision and only occasionally added. Obviously a poorly implemented Engine implementation could add excessive user-data which could cause issues. But it is an Engine implementation, it has full access to everything (memory, disk, etc) so it generally trusted.